frontends/web/src/components/wait-dialog/wait-dialog.tsx

```diff
@@ -14,66 +15,94 @@
  * limitations under the License.
  */
 
-import { Component, h } from 'preact';
-import { translate } from 'react-i18next';
+import { Component, ComponentChild, h, JSX, RenderableProps } from 'preact';
+import { translate, TranslateProps } from '../../decorators/translate';
 import approve from '../../assets/icons/hold.png';
 import reject from '../../assets/icons/tap.png';
 import { animate } from '../../utils/animation';
 import * as style from '../dialog/dialog.css';
 
-@translate()
-export default class WaitDialog extends Component {
-    state = {
+interface WaitDialogProps {
+    includeDefault?: boolean;
+    prequel?: JSX.Element | false;
```

Comment:
why `| false`?

`undefined` should be enough?

Reply:
TS said it can be false :)
but looking into it more closely:

The only place that `prequel` is used is here https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/routes/account/send/send.tsx#L747

The variable is set here https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/routes/account/send/send.tsx#L575
and if `signProgress.steps > 1` is 1 or less the variable will be `false`.
`true` wouldn't make sense, but I'd expect `false` to do the same as undefined just falsy :).

Should I rewrite send.tsx a bit, so it would be either a JSX.Element or undefined but not false?

Tho I'd prefer in this case `prequel?: JSX.Element | null;` so we could set the variable `condition ? (<span>) : null` 
 but that is a matter of taste we already use `undefined` in various places.

Reply:
I still think anything beyond `undefined | Element` is too many needless states. send.tsx can just set it to undefined if it is not used.

undefined vs null: no strong opinion, but undefined seems to be the convention in typescript.